mqtt/mqttd/src/broker/mod.rs

```diff
@@ -13,10 +13,17 @@ use tokio::{
 use tracing::{info, warn};
 
 use mqtt_broker::{
-    BrokerHandle, FilePersistor, Message, Persist, ShutdownHandle, Snapshotter,
-    StateSnapshotHandle, SystemEvent, VersionedFileFormat,
+    BrokerHandle, FilePersistor, Message, Persist, ShutdownHandle as SnapshotShutdownHandle,
+    Snapshotter, StateSnapshotHandle, SystemEvent, VersionedFileFormat,
 };
 
+#[cfg(feature = "edgehub")]
```

Comment:
Note: These cfg do not currently work to support compiling with the generic flag. There will be a second PR to support sidecars that will fix this.